### DISCUSSION TITLE
Macbook unable to connect to RPi wifi using RaspAP

### DISCUSSION THREAD

**Discussion by gainovermg** [RaspAP/raspap-webgui#996] [Q&A]
Just hoping to get some direction.

I'm trying to build a Wireless AP using RaspAP and a Pi 3B. Everything seems to work fine, except for one little issue: I can't get one machine(a Macbook Pro) to connect to the AP. I've tried a Dell Laptop, iPhone, Nintendo Switch - all of these will connect. Every time I try to connect the Mac, I get the following error: failed to send packet, permission denied.

```
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 available DHCP range: 10.3.141.50 -- 10.3.141.255
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 client provides name: Marks-MBP
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 DHCPREQUEST(wlan0) 10.3.141.255 a4:5e:60:d3:1d:b5
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 tags: wlan0
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 DHCPACK(wlan0) 10.3.141.255 a4:5e:60:d3:1d:b5 Marks-MBP
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 requested options: 1:netmask, 121:classless-static-route, 3:router,
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 requested options: 6:dns-server, 15:domain-name, 119:domain-search,
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 requested options: 252, 95, 44:netbios-ns, 46:netbios-nodetype
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 next server: 10.3.141.1
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 sent size: 1 option: 53 message-type 5
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 sent size: 4 option: 54 server-identifier 10.3.141.1
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 sent size: 4 option: 51 lease-time 12h
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 sent size: 4 option: 58 T1 6h
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 sent size: 4 option: 59 T2 10h30m
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 sent size: 4 option: 1 netmask 255.255.255.0
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 sent size: 4 option: 28 broadcast 10.3.141.255
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 sent size: 4 option: 3 router 10.3.141.1
Sep 2 09:10:34 dnsmasq-dhcp[966]: 2457421306 sent size: 4 option: 6 dns-server 10.3.141.1
Sep 2 09:10:34 dnsmasq[966]: failed to send packet: Permission denied
```

Really just curious what permissions are being denied, from where, and how to correct.

**Comment by billz** [comment]
I also use a MBP but have not observed this issue. If you're running an instance of dnsmasq on your laptop this may create conflicts. 

If not, you could trace system calls on the RPi and try to get more detail. Use `ps aux |grep dnsmasq` to get the PID, then `sudo strace -e trace=%network -p [PID]`. Connect your MBP again and look for additional insights not provided by your logs.

You could troubleshoot this until the cows come home or, alternatively, use the workaround [in this FAQ](https://docs.raspap.com/faq/#clients-cannot-obtain-an-ip-address-from-the-ap):

> As a last resort, you can assign a static IP address to your device. Copy the MAC address for your device as it appears above and create a new entry in RaspAP's **DHCP Server > Static Leases** tab. Save settings, restart dnsmasq and try connecting your client again.
